### PR TITLE
Titan Notches for #2558

### PR DESCRIPTION
**Pull request by bobgeorgethe3rd**
As mentioned in #2558 this update adds the following notches.

- Cross and Crotch Seam notches
- Hip notches (when the rise is high enough)
- Seat Notches
- Knee notches

Changed logoAnchor.y to fork.y to minimise overlap of new paths.

Another thing I added was Hip Line and Seat Line paths to better identify where the hip line is with the notches.

I have also changed the waistbandWidth option to allow 0%, whilst the waistband being included is awesome for extending it is not the best for those who are using Titan as a physical block or for those who are testing Titan to check their measures.
Whilst not strictly necessary for extending Titan as some of these notches may be in a different place when extended, it will benefit those who are using Titan as it is described in its instructions. 

This has made me think about another possible addition to Titan which will benefit those extending it so may be another update soon. 

Added options.fitGuides so it does not display on Charlie and Paco.

Edit: Added seat notches.
Edit2: Added fitGuides

### CODE REVIEW

**Comment by bobgeorgethe3rd** [comment]
Ok turns out there was some confusion over hip and seat.
I will be adding seat notches to this so have converted to a draft for the minute.

**Comment by bobgeorgethe3rd** [comment]
Added.

**Comment by bobgeorgethe3rd** [comment]
Crap just checked on gitpod need to add an option to turn it off in extended patterns

**Comment by bobgeorgethe3rd** [comment]
Had to do it manually because my permissions need a look into annoyingly.

File: designs/titan/src/back.js [hunk 1]
```diff
@@ -221,15 +221,80 @@ export default (part) => {
...
+    points.logoAnchor = new Point(points.crossSeamCurveStart.x / 2, points.fork.y)
     snippets.logo = new Snippet('logo', points.logoAnchor)
     points.titleAnchor = points.logoAnchor.shift(-90, 60)
     macro('title', {
       nr: 1,
       title: 'back',
       at: points.titleAnchor,
     })
-
+	//notches
```
Comment (BenJamesBen) [inline designs/titan/src/back.js:232]: For this entire section of new code, indentation is missing. While not technically necessary, I recommend/request that indentation be added for readability and maintainability.

File: designs/titan/src/front.js [hunk 2]
```diff
@@ -295,15 +295,78 @@ export default (part) => {
...
+	on: ['hipsIn', 'hipsOut',]
+	})
+	paths.hipline = new Path()
+	.move(points.hipsOut)
+	.line(points.hipsIn)
+	.attr('class', 'fabric help')
+	.attr('data-text', 'Hip Line')
+	.attr('data-text-class', 'center')
+	}
+}
```
Comment (BenJamesBen) [inline designs/titan/src/front.js:369]: Similarly, in this whole section I found it very difficult to read the non-indented code to determine what the code was supposed to do. Please add indentation.

File: designs/titan/src/back.js [hunk 3]
```diff
@@ -221,15 +221,80 @@ export default (part) => {
...
+	points.kneeInNotch = utils.lineIntersectsCurve(points.kneeIn, points.kneeOut.rotate(180, points.kneeIn), points.fork, points.forkCp2, points.kneeInCp1, points.floorIn)
+	}
+	macro('sprinkle', {
+	snippet: 'notch',
+	on: ['seatOutNotch', 'kneeInNotch', 'kneeOutNotch']
+	})
+	macro('sprinkle', {
+	snippet: 'bnotch',
+	on: ['crossSeamCurveStart', 'seatIn']
+	})
```
Comment (BenJamesBen) [inline designs/titan/src/back.js:280]: On the Back, you are using bnotches for these two points. However, on the Front you are using regular notches for the same points. Is this intended?

**Review by BenJamesBen — CHANGES_REQUESTED** [review]

**Comment by BenJamesBen** [comment]
FYI, this is what the added help lines and notches look like on the upper sections of the parts.
Screen Shot 2022-08-12 at 12 30 50 PM

File: designs/titan/src/back.js (hunk 3 above)
Comment (bobgeorgethe3rd) [reply designs/titan/src/back.js:280]: Yes it identifies the cross seam. This seam is connected back to back whereas the others are connected to the front. One of the requests was to add notch identifiers to the back. This fufills that request.

File: designs/titan/src/back.js (hunk 1 above)
Comment (bobgeorgethe3rd) [reply designs/titan/src/back.js:232]: I have made this changed but personally I find the opposite to be true.

File: designs/titan/src/front.js (hunk 2 above)
Comment (bobgeorgethe3rd) [reply designs/titan/src/front.js:369]: I have made this changed but personally I find the opposite to be true.

**Comment by bobgeorgethe3rd** [comment]
> FYI, this is what the added help lines and notches look like on the upper sections of the parts. Screen Shot 2022-08-12 at 12 30 50 PM
> 
> I tested the changes in the lab, and the actual notch and helpline placements seem correct.

I did test in the lab and had realised I forgot to check Charlie and Paco. I was working on the local end when making the changes hence my misstep. This change would be very difficult to do without visual representation.

File: designs/titan/src/back.js (hunk 3 above)
Comment (BenJamesBen) [reply designs/titan/src/back.js]: Okay. I still don't understand why crossSeamCurveStart and seatIn are bnotches instead of notches, and why so only on the back but not the front. However, if you say that this arrangement is intentional, then I am fine with it.

**Review by BenJamesBen — COMMENTED** [review]
The indentation is still not quite correct. (All the nestled if-else statements are incorrectly at the same indentation level. And, you start your indentation at level 1 but incorrectly end up at level 0.) And, I suspect that some code lines are at the wrong indentation levels as a result. But, the code is now easier to read, where it is easier to see where the indentation errors are, so thank you for making the changes.

As I mentioned in a comment buried within the code review, I still don't quite understand why those two notches are bnotches instead of regular notches, and why so only on the back but not the front. However, since you said that this arrangement is intentional, I am okay with it.

File: designs/titan/config/index.js [hunk 4]
```diff
@@ -71,6 +72,6 @@ export default {
     crotchSeamCurveAngle: { deg: 25, min: 0, max: 35 },
     waistBalance: { pct: 60, min: 30, max: 90 },
     grainlinePosition: { pct: 45, min: 30, max: 60 },
-    waistbandWidth: { pct: 3, min: 1, max: 6, snap: elastics, ...pctBasedOn('waistToFloor') },
+    waistbandWidth: { pct: 3, min: 0, max: 6, snap: elastics, ...pctBasedOn('waistToFloor') },
```
Comment (joostdecock) [inline designs/titan/config/index.js:75]: This seems out of scope for this PR

**Review by joostdecock — CHANGES_REQUESTED** [review]
Please run `yarn prettier` in `designs/titan` then submit the changes.

**Comment by joostdecock** [comment]
To be clear, @BenJamesBen is correct that the indentation is a bit all over the place, making it hard to read.

However, you don't have to worry about code formatting.  Just run `yarn prettier` and it will be done for you.

File: designs/titan/config/index.js (hunk 4 above)
Comment (bobgeorgethe3rd) [reply designs/titan/config/index.js:75]: I thought it would be useful but I will change it back

**Comment by bobgeorgethe3rd** [comment]
Figured out what my gitpod issue was :) Thank you for telling me about `yarn prettier`. I had no idea how to fix the indentation in the way needed.

**Review by BenJamesBen — APPROVED** [review]
The revised code changes look fine.

(I was wondering why you reverted the waistband minimum back to 1%, but for the purpose of this code review it doesn't matter.)

**Review by joostdecock — APPROVED** [review]
